### PR TITLE
Fix has_table_privilege for unknown tables

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Continuation from https://github.com/crate/crate/pull/16004. `has_table_privilege` is new to `5.8`, no need backporting.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
BTW, `has_database_privilege` throws exceptions for unknown databases but it is not info leak since we state in docs that CrateDB supports `crate` only:

https://cratedb.com/docs/crate/reference/en/latest/general/builtins/scalar-functions.html#current-database
> The current_database function returns the name of the current database, which in CrateDB will always be crate:

File: server/src/test/java/io/crate/expression/scalar/HasTablePrivilegeFunctionTest.java [hunk 1]
```diff
@@ -121,4 +122,51 @@ public void test_throws_if_listed_privileges_contain_invalid_privilege() {
...
+        Privilege create4 = new Privilege(Policy.GRANT, Permission.DDL, Securable.SCHEMA, "my_schema", "crate");
+        var user4 = RolesHelper.userOf("test", Set.of(create4), null);
+        sqlExpressions = new SqlExpressions(tableSources, null, user4, List.of(), schemas);
+        assertEvaluate("has_table_privilege('my_schema.unknown_table', 'SELECT')", false);
+        assertEvaluate("has_table_privilege(" + tableOid + ", 'SELECT')", false);
+
+        // returns false otherwise
+        sqlExpressions = new SqlExpressions(tableSources, null, TEST_USER_WITH_DOC_USERS_TABLE_DQL, List.of(), schemas);
+        assertEvaluate("has_table_privilege('my_schema.unknown_table', 'SELECT')", false);
+        assertEvaluate("has_table_privilege(" + tableOid + ", 'SELECT')", false);
```
[the remark this answers is not in the document]
Comment (jeeminso) [reply server/src/test/java/io/crate/expression/scalar/HasTablePrivilegeFunctionTest.java:170]: Please check updates if I understood correctly.

File: server/src/main/java/io/crate/expression/scalar/HasTablePrivilegeFunction.java [hunk 2]
```diff
@@ -51,10 +51,12 @@ public static boolean checkByTableName(Roles roles, Role user, Object table, Col
...
         int tableOid = (int) table;
         RelationName relationName = schemas.getRelation(tableOid);
-        if (relationName == null) {
-            throw new IllegalArgumentException("Cannot find corresponding relation by the given oid");
-        }
-        String tableFqn = relationName.fqn();
+
+        // Proceed to checkPrivileges with tableIdent as 'null' which will return 'true' for a superuser or a user with
+        // appropriate cluster scope privileges.
+        // Note that a user with schema privileges will always get a 'false' since we cannot identify the schema name from the
```
[the remark this answers is not in the document]
Comment (jeeminso) [reply server/src/main/java/io/crate/expression/scalar/HasTablePrivilegeFunction.java]: Clarified the comments, please check updated commit.

**Review by matriv — APPROVED** [review]
thx!

**Pull request merged**